src/v2/Apps/Order/Components/__tests__/SavedAddresses.jest.tsx

```diff
@@ -1,9 +1,113 @@
 import { graphql } from "react-relay"
 import { SavedAddressesFragmentContainer } from "../SavedAddresses"
 import { setupTestWrapper } from "v2/DevTools/setupTestWrapper"
+import { Button } from "@artsy/palette"
+import { AddressModal } from "v2/Apps/Order/Components/AddressModal"
+import React from "react"
 
 jest.unmock("react-relay")
 
+beforeEach(() => {
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+})
+
+describe("SavedAddress button interactions", () => {
+  const { getWrapper } = setupTestWrapper({
+    Component: (props: any) => (
+      <SavedAddressesFragmentContainer inCollectorProfile {...props} />
+    ),
+    query: graphql`
+      query SavedAddresses_Test_Query {
+        me {
+          ...SavedAddresses_me
+        }
+      }
+    `,
+  })
+
+  const wrapper = getWrapper({
+    Me: () => ({
+      addressConnection: mockAddressConnection,
+    }),
+  })
+
+  it("renders modal when button is clicked", () => {
+    const button = wrapper.find(Button)
+    const modal = wrapper.find(AddressModal)
+    expect(modal).toHaveLength(0)
+    button.props().onClick({} as any)
+
+    setTimeout(() => {
+      expect(modal).toHaveLength(1)
+    }, 0)
+  })
+
+  it("opens Create Address modal when as expected", () => {
+    const button = wrapper.find(Button)
+    const modal = wrapper.find(AddressModal)
+    expect(modal).toHaveLength(0)
+    button.props().onClick({} as any)
+
+    setTimeout(() => {
+      expect(modal.props().modalDetails).toBe({
+        addressModalTitle: "Add new address",
+        addressModalAction: "createUserAddress",
+      })
+    }, 0)
+  })
+})
+
+describe("SavedAddress", () => {
+  const { getWrapper } = setupTestWrapper({
+    Component: (props: any) => <SavedAddressesFragmentContainer {...props} />,
+    query: graphql`
+      query SavedAddresses_Test_Query {
```

Comment:
Relay doesn't like this and throws `Duplicate definition of 'SavedAddresses_Test_Query'` error

Reply:
Fixed it!